### ISSUE TITLE
Build order issue using a cmake subproject dependency

### ISSUE THREAD

**Issue by EmmanuelP** [mesonbuild/meson#9963]
### Discussed in https://github.com/mesonbuild/meson/discussions/9935

<div type='discussions-op-text'>

<sup>Originally posted by **EmmanuelP** February  2, 2022</sup>
Hi,

Before reporting it as an issue in meson, I'd like to know if I'm doing it wrong.

I have a pretty simple repository with open62541 dependency as a subproject. open62541 build system is based on cmake. The meson part is the following:

``` meson
open62541 = cmake.subproject('open62541', options:open62541_options)
opc_dep = open62541.dependency('open62541')

opctest_dependencies = [opc_dep]

apps = ['producer', 'consumer']

foreach a: apps
  executable(a, a + '.c', dependencies: opctest_dependencies, install: true)
endforeach
```

The problem is when I invoke ninja, the two executables are always built before the open62541 subproject. Is there something wrong in the way I have declared the dependencies ?

The full project is here: [https://github.com/EmmanuelP/opctest](https://github.com/EmmanuelP/opctest)</div>

To summarize, I expect a dependency created from a cmake subproject to be built before the executables that depend on it.
